pkg/kubelet/apis/config/validation/validation_test.go

```diff
@@ -377,286 +377,331 @@ func TestValidateKubeletConfiguration(t *testing.T) {
-		},
-		errMsg: "invalid configuration: containerLogMonitorInterval must be a positive time duration greater than or equal to 3s",
-	}, {
-		name: "containerLogMonitorInterval must be at least 3s or higher",
-		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			conf.ContainerLogMonitorInterval = metav1.Duration{Duration: 2 * time.Second}
+			i := int32(0)
+			conf.FeatureGates = map[string]bool{"EnableKubeletCrashLoopBackoffMax": true}
+			conf.CrashLoopBackOff = &kubeletconfig.CrashLoopBackOffConfig{
+				MaxSeconds: &i,
+			}
 			return conf
 		},
-		errMsg: "invalid configuration: containerLogMonitorInterval must be a positive time duration greater than or equal to 3s",
-	}, {
-		name: "pod logs path must be not empty",
-		configure: func(config *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			config.PodLogsDir = ""
-			return config
-		},
-		errMsg: "invalid configuration: podLogsDir was not specified",
-	}, {
-		name: "pod logs path must be absolute",
-		configure: func(config *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			config.PodLogsDir = "./test"
-			return config
-		},
-		errMsg: `invalid configuration: pod logs path "./test" must be absolute path`,
+		errMsg: "invalid configuration: crashloopbackoff.maxSeconds (got: 0) must be set between 1 and 300",
 	}, {
-		name: "pod logs path must be normalized",
-		configure: func(config *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			config.PodLogsDir = "/path/../"
-			return config
-		},
-		errMsg: `invalid configuration: pod logs path "/path/../" must be normalized`,
-	}, {
-		name: "pod logs path is ascii only",
-		configure: func(config *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			config.PodLogsDir = "/🧪"
-			return config
-		},
-		errMsg: `invalid configuration: pod logs path "/🧪" mut contains ASCII characters only`,
-	}, {
-		name: "invalid ContainerRuntimeEndpoint",
+		name: "CrashLoopBackOff.MaxSeconds too high",
 		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			conf.ContainerRuntimeEndpoint = ""
+			i := int32(301)
+			conf.FeatureGates = map[string]bool{"EnableKubeletCrashLoopBackoffMax": true}
+			conf.CrashLoopBackOff = &kubeletconfig.CrashLoopBackOffConfig{
+				MaxSeconds: &i,
+			}
 			return conf
 		},
-		errMsg: "invalid configuration: the containerRuntimeEndpoint was not specified or empty",
+		errMsg: "invalid configuration: crashloopbackoff.maxSeconds (got: 301) must be set between 1 and 300",
 	}, {
-		name: "invalid Logging configuration",
+		name: "CrashLoopBackOff.MaxSeconds ok",
 		configure: func(conf *kubeletconfig.KubeletConfiguration) *kubeletconfig.KubeletConfiguration {
-			conf.Logging.Format = "invalid"
+			i := int32(3)
+			conf.FeatureGates = map[string]bool{"EnableKubeletCrashLoopBackoffMax": true, "CustomCPUCFSQuotaPeriod": true}
```

Comment:
Why is the `CustomCPUCFSQuotaPeriod` FeatureGate also set to `true` here?

Reply:
Something seems to be up with the validation logic for `CustomCPUCFSQuotaPeriod` when the feature gate is not set, I think it's something weird with https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/validation/validation.go#L71 and its defaults either generally or as computed by this test. I am looking into it more seriously now.